src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,101 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import classNames from 'classnames';
+import React, { useState } from 'react';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todo: Todo,
+  onUpdate(id: number, todo: Todo | null): void,
+};
+
+export const TodoItem: React.FC<Props> = ({ todo, onUpdate }) => {
+  const [title, setTitle] = useState(todo.title);
+  const [editing, setEditing] = useState(false);
+
+  const editTodo = (count: number) => {
+    if (count === 2) {
+      setEditing(true);
+    }
+  };
+
+  const handleOnEdit = (typeOfValue: string, value: string | boolean) => {
+    let newTodo: Todo | null = { ...todo };
+
+    switch (typeOfValue) {
+      case 'title':
+        if (typeof value === 'string') {
+          if (value.length === 0) {
+            handleOnEdit('deleted', true);
```

Comment:
are you sure you couldn't avoid implementing such a complicated logic?
function calls itself because `value` has two types – it's a crutch